lib/propolis/src/hw/bhyve/pmtimer.rs

```diff
@@ -1,40 +1,86 @@
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
 
 use crate::inventory::Entity;
 use crate::migrate::*;
 use crate::vmm::VmmHdl;
 
 pub struct BhyvePmTimer {
     hdl: Arc<VmmHdl>,
+    inner: Mutex<Inner>,
+}
+struct Inner {
+    ioport: u16,
+    attached_ioport: Option<u16>,
 }
 impl BhyvePmTimer {
-    pub fn create(hdl: Arc<VmmHdl>) -> Arc<Self> {
-        Arc::new(Self { hdl })
+    pub fn create(hdl: Arc<VmmHdl>, ioport: u16) -> Arc<Self> {
+        Arc::new(Self {
+            hdl,
+            inner: Mutex::new(Inner { ioport, attached_ioport: None }),
+        })
+    }
+    fn update_attachment(&self) {
+        let mut inner = self.inner.lock().unwrap();
+
+        let target = inner.ioport;
+        let exists = inner.attached_ioport.as_ref();
+        if matches!(exists, Some(p) if *p == target) {
+            // Attachment is already correct
+            return;
+        }
+
+        match self.hdl.pmtmr_locate(target) {
+            Ok(()) => {
+                inner.attached_ioport = Some(target);
+            }
+            Err(_e) => {
+                inner.attached_ioport = None;
+                // TODO: squawk about it?
```

Comment:
I left it this way, since there's nothing actionable to do in the `resume()` call site for now.  It could be something to revisit if/when the IO port mapping stuff is improved.